packages/common-utils/src/metadata.ts

```diff
@@ -426,10 +428,12 @@ export class Metadata {
         query: sql.sql,
         query_params: sql.params,
         connectionId: chartConfig.connection,
-        clickhouse_settings: {
-          max_rows_to_read: DEFAULT_SAMPLE_SIZE,
-          read_overflow_mode: 'break',
-        },
+        clickhouse_settings: !disableRowLimit
```

Comment:
I'm a bit worried about the performance impact. I wonder if wa want to enable caching for this method (using `this.cache.getOrFetch`)